File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/ForegroundUpdateTimeoutModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable;

import org.threeten.bp.Duration;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;
import dagger.Reusable;

import static se.gustavkarlsson.skylight.android.dagger.Names.FOREGROUND_UPDATE_TIMEOUT_NAME;

@Module
public abstract class ForegroundUpdateTimeoutModule {

	// Published
	@Provides
	@Reusable
	@Named(FOREGROUND_UPDATE_TIMEOUT_NAME)
	static Duration provideForegroundUpdateTimeout() {
		return Duration.ofSeconds(10);
	}

}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/ForegroundUpdateTimeoutModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable

import dagger.Module
import dagger.Provides
import dagger.Reusable
import org.threeten.bp.Duration
import se.gustavkarlsson.skylight.android.dagger.FOREGROUND_UPDATE_TIMEOUT_NAME
import javax.inject.Named

@Module
class ForegroundUpdateTimeoutModule {

    // Published
    @Provides
    @Reusable
    @Named(FOREGROUND_UPDATE_TIMEOUT_NAME)
    fun provideForegroundUpdateTimeout(): Duration {
        return Duration.ofSeconds(10)
    }

}
